pkg/providers/tinkerbell/reconciler/reconciler_test.go

```diff
@@ -91,6 +92,79 @@ func TestReconcileCNIErrorClientRegistry(t *testing.T) {
 	tt.Expect(result).To(Equal(controller.Result{}))
 }
 
+func TestReconcilerValidateClusterSpecInvalidDatacenterConfig(t *testing.T) {
```

Comment:
If all of these are properly tested in their original package, I don't think we need to test all of the cases here. One negative and one positive is enough.

Reply:
yeah got a little carried away. kept only a couple test cases now.